src/functionaltests/java/com/ericsson/ei/subscriptions/content/SubscriptionContentSteps.java

```diff
@@ -0,0 +1,158 @@
+package com.ericsson.ei.subscriptions.content;
+
+import com.ericsson.ei.controller.model.Subscription;
+import com.ericsson.ei.controller.model.SubscriptionResponse;
+import com.ericsson.ei.utils.FunctionalTestBase;
+import com.ericsson.ei.utils.HttpRequests;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Ignore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import javax.annotation.PostConstruct;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+
+
+@Ignore
+public class SubscriptionContentSteps extends FunctionalTestBase {
+
+    @LocalServerPort
+    int applicationPort;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionContentSteps.class);
+    private HttpRequests request;
+    private ResponseEntity response;
+
+    private String validSubscriptionFile = "src/functionaltests/resources/ValidSubscription.json";
+    private String invalidSubscriptionFile = "src/functionaltests/resources/InvalidSubscription.json";
+
+    ObjectMapper mapper = new ObjectMapper();
+
+    @PostConstruct
+    private void setUp() {
+        request = new HttpRequests(applicationPort);
+    }
+
+
+    // SCENARIO 1
+
+
+    @Given("^No subscriptions exist$")
+    public void fetch_subscriptions() {
+        String url = "http://localhost:" + applicationPort + "/subscriptions";
+        response = request.makeHttpGetRequest(url);
+        assertEquals("[]", response.getBody().toString());
+    }
+
+    @When("^Create subscription request$")
+    public void send_subscription_request() {
+        String url = "http://localhost:" + applicationPort + "/subscriptions";
+        response = request.makeHttpPostRequest(url, validSubscriptionFile);
+    }
+
+    @Then("^The subscription is created successfully$")
+    public void subscription_created_successfully() {
+        assertEquals(HttpStatus.OK.value(), response.getStatusCodeValue());
```

Comment:
Can't response object be an input parameter to this function,,, or is it Cucumber framework that forces us to use class global response object.

Reply:
The way the scenarios are written I split up making the request and asserting the correct
reponse, and cucumber takes care of calling the tests methods sequentially. Therefore I made a
global response object which could be accessed from any test method inside the step file.

Instead of explicitly calling each test method and clutter the step file, Cucumber takes care
of executing the scenarios sequentially in the defined order in the feature file, which I rather prefer. 

Using Cucumber framework we can group test steps together and execute full scenarios. 
An advantage of what you are proposing might be to run tests in parallel, but if that's what we need/want 
it is also possible to split scenarios up between different features as we've decided to run each feature 
in parallel.